ql/pricingengines/bond/bondfunctions.hpp

```diff
@@ -153,6 +155,32 @@ namespace QuantLib {
                           Real accuracy = 1.0e-10,
                           Size maxIterations = 100,
                           Rate guess = 0.05);
+        template <typename Solver>
+        static Rate yield(const Bond& bond,
```

Comment:
I think this (and also `CashFlows::yield`) should take the solver as an argument.  This way, it would be possible to set other parameters to the solver, as well as to use solvers that don't have a default constructor.